README.md
# https-www.pythonanywhere.com-user-gianfrancodeluca-consoles-10020075-


history.txt
    1  ls
    2  cd mysite
    3  ls
    4  git config --global user.name "gianfrancodeluca"
    5  git config --global user.email "rmike8328@gmail.com"
    6  git init
    7  cat > .gitignore
    8  git status
    9  git add .
   10  git commit -m "include commits"
   11  git status
   12  git add .
   13  git commit -m "[master (root-commit) dd4545]"
   14  git status
   15  git add .
   16  git commit -m "after the code you just added"
   17  git status
   18  git add .
   19  git commit -m "9 insertions (+)"
   20  ipython3.6
   21  git status
   22  git add .
   23  git commit -m "git commit -a"
   24  git diff
   25  git status
   26  git add .
   27  git commit -m "/main_page.html"
   28  git status
   29  git add .
   30  git commit -m "git add <file>"
   31  mkvirtualenv flask-tutorial --python=python3.6
   32  pip install flask flask-login flask-sqlalchemy mysql-connector-python
   33  pip freeze > requirements.txt
   34  git status
   35  git add .
   36  git commit -m "click-6.7"
   37  #migrations
   38  #pip install flask migrate
   39  pip install Flask-Migrate
   40  export FLASK_APP=flask_app.py
   41  flask db init
   42  flask db migrate
   43  git status
   44  git add .
   45  git commit -m "seize martyres decapites"
   46  flask db stamp head
   47  pip freeze > requirements.txt
   48  git status
   49  git add .
   50  git commit -m "un petit peu moins"
   51  flask db migrate
   52  flask db upgrade
   53  git status
   54  git add.
   55  git add .
   56  git commit -m "faites connaissance avec julien"
   57  flask db migrate
   58  flask db upgrade
   59  flask shell
   60  git status
   61  flask db migrate
   62  flask db upgrade
   63  flask shell
   64  echo "# https-www.pythonanywhere.com-user-gianfrancodeluca-consoles-10020075-" >> README.md
   65  git init
   66  git add README.md
   67  git commit -m "first commit"
   68  git remote add origin https://github.com/mikerissi/https-www.pythonanywhere.com-user-gianfrancodeluca-consoles-10020075-.git
   69  git push -u origin master
   70  history
   71  history > history.txt


flask_app.py
from datetime import datetime
from flask import Flask, redirect, render_template, request, url_for
from flask_login import current_user, login_required, login_user, LoginManager, logout_user, UserMixin
from flask_migrate import Migrate
from flask_sqlalchemy import SQLAlchemy
from werkzeug.security import check_password_hash


app = Flask(__name__)
app.config["DEBUG"] = True

SQLALCHEMY_DATABASE_URI = "mysql+mysqlconnector://{username}:{password}@{hostname}/{databasename}".format(
    username="gianfrancodeluca",
    password="Exeter22",
    hostname="gianfrancodeluca.mysql.pythonanywhere-services.com",
    databasename="gianfrancodeluca$comments",
)
app.config["SQLALCHEMY_DATABASE_URI"] = SQLALCHEMY_DATABASE_URI
app.config["SQLALCHEMY_POOL_RECYCLE"] = 299
app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
db = SQLAlchemy(app)
migrate = Migrate(app, db)

app.secret_key = "65465321332989898"
login_manager = LoginManager()
login_manager.init_app(app)


class User(UserMixin, db.Model):

    __tablename__ = "users"

    id = db.Column(db.Integer, primary_key=True)
    username = db.Column(db.String(128))
    password_hash = db.Column(db.String(128))

    def check_password(self, password):
        return check_password_hash(self.password_hash, password)


    def get_id(self):
        return self.username


@login_manager.user_loader
def load_user(user_id):
    return User.query.filter_by(username=user_id).first()


class Comment(db.Model):

    __tablename__ = "comments"

    id = db.Column(db.Integer, primary_key=True)
    content = db.Column(db.String(4096))
    posted = db.Column(db.DateTime, default=datetime.now)
    commenter_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=True)
    commenter = db.relationship('User', foreign_keys=commenter_id)


@app.route("/", methods=["GET", "POST"])
def index():
    if request.method == "GET":
        return render_template("main_page.html", comments=Comment.query.all())

    if not current_user.is_authenticated:
        return redirect(url_for('index'))

    comment = Comment(content=request.form["contents"], commenter=current_user)
    db.session.add(comment)
    db.session.commit()
    return redirect(url_for('index'))



@app.route("/login/", methods=["GET", "POST"])
def login():
    if request.method == "GET":
        return render_template("login_page.html", error=False)

    user = load_user(request.form["username"])
    if user is None:
        return render_template("login_page.html", error=True)

    if not user.check_password(request.form["password"]):
        return render_template("login_page.html", error=True)

    login_user(user)
    return redirect(url_for('index'))



@app.route("/logout/")
@login_required
def logout():
    logout_user()
    return redirect(url_for('index'))

migrations/versions/dc0c1c639278_.py
"""empty message

Revision ID: dc0c1c639278
Revises: 
Create Date: 2018-08-30 23:44:10.258443

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = 'dc0c1c639278'
down_revision = None
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('users',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('username', sa.String(length=128), nullable=True),
    sa.Column('password_hash', sa.String(length=128), nullable=True),
    sa.PrimaryKeyConstraint('id')
    )
    op.create_table('comments',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('content', sa.String(length=4096), nullable=True),
    sa.Column('posted', sa.DateTime(), nullable=True),
    sa.Column('commenter_id', sa.Integer(), nullable=True),
    sa.ForeignKeyConstraint(['commenter_id'], ['users.id'], ),
    sa.PrimaryKeyConstraint('id')
    )
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_table('comments')
    op.drop_table('users')
    # ### end Alembic commands ###


migrations/versions/6a0e31187dc3_.py
"""empty message

Revision ID: 6a0e31187dc3
Revises: dc0c1c639278
Create Date: 2018-08-30 23:55:15.457494

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = '6a0e31187dc3'
down_revision = 'dc0c1c639278'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('users',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('username', sa.String(length=128), nullable=True),
    sa.Column('password_hash', sa.String(length=128), nullable=True),
    sa.PrimaryKeyConstraint('id')
    )
    op.add_column('comments', sa.Column('commenter_id', sa.Integer(), nullable=True))
    op.add_column('comments', sa.Column('posted', sa.DateTime(), nullable=True))
    op.create_foreign_key(None, 'comments', 'users', ['commenter_id'], ['id'])
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_constraint(None, 'comments', type_='foreignkey')
    op.drop_column('comments', 'posted')
    op.drop_column('comments', 'commenter_id')
    op.drop_table('users')
    # ### end Alembic commands ###
